django/utils/log.py

```diff
@@ -189,3 +192,43 @@ def format(self, record):
 
         record.msg = msg
         return super(ServerFormatter, self).format(record)
+
+
+def log_response(message, *args, **kwargs):
+    """Logs errors based on response status.
+
+    Log 5xx responses in error level and 4xx in warning level (unless level
+    is explicitly given as a keyword argument). The Response status_code and
+    the entire request are passed to logger extra parameter. This function
+    also avoid duplicated logs.
+
+    """
+
+    for argname in ('request', 'response'):
+        if argname not in kwargs:
+            raise TypeError("'log_response' call missing the required keyword argument: '{}'".format(argname))
+
+    response = kwargs.pop('response')
```

Comment:
I'm tempted to suggest making `response` the first argument (before `message`), making the calling signature more concise and allowing Python to handle it as a normal required argument. The downside of course is that it makes the signature less like a normal logging call. I don't have strong feelings, ok with however you prefer to paint this bikeshed.